src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -55,6 +63,95 @@ const ApplicationLauncher = _.flow(
   const runtime = getCurrentRuntime(runtimes)
   const runtimeStatus = getConvertedRuntimeStatus(runtime) // preserve null vs undefined
 
+  const FileOutdatedModal = ({ onDismiss, bucketName }) => {
+    const handleChoice = _.flow(
+      withErrorReportingInModal('Error setting up analysis file syncing')(onDismiss),
+      Utils.withBusyState(setBusy)
+    )(async shouldCopy => {
+      await Promise.all(_.flatMap(async analysis => {
+        const currentMetadata = analysis.metadata
+        const file = getFileName(analysis.name)
```

Comment:
FYI that you can write the above as

```suggestion
      await Promise.all(_.flatMap(async ({ name, metadata: currentMetadata }) => {
        const file = getFileName(name)
```
which helps with concision. Not a blocker though.

Reply:
i like this pattern more, updated my code!